main/main.go

```diff
@@ -0,0 +1,226 @@
+package main
+
+import (
+	"bufio"
+	"github.com/stackrox/CvesPusher/utils"
+	"io/ioutil"
+
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/clair/pkg/commonerr"
+	"github.com/coreos/clair/pkg/httputil"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	dataFeedURL     string = "https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-%s.json.gz"
+	dataFeedMetaURL string = "https://nvd.nist.gov/feeds/json/cve/1.0/nvdcve-1.0-%s.meta"
+	logDataFeedName string = "data feed name"
+	filePrefix      string = "NVD"
+	kubernetes      string = "k8s"
+	istio           string = "istio"
+	fileExt         string = ".json"
+	cves            string = "CVEs"
+)
+
+type appender struct {
+	localPath      string
+	dataFeedHashes map[string]string
+	metadata       map[string]NVDMetadata
+}
+
+type NVDMetadata struct {
+	CVSSv2 NVDmetadataCVSSv2
+	CVSSv3 NVDmetadataCVSSv3
+}
+
+type NVDmetadataCVSSv2 struct {
+	PublishedDateTime string
+	Vectors           string
+	Score             float64
+}
+
+type NVDmetadataCVSSv3 struct {
+	Vectors             string
+	Score               float64
+	ExploitabilityScore float64
+	ImpactScore         float64
+}
+
+func main() {
+	fileMap := make(map[string]string)
+
+	path, err := utils.IsEnvVarNonEmpty("LOCAL_FILE_PATH")
+	if err != nil {
+		log.Fatal(err)
+		os.Exit(1)
+	}
+
+	err = utils.IsPathOk(path)
+	if err != nil {
+		log.Error(err)
+		os.Exit(1)
+	}
+
+	feedReader, _, err := getDataFeeds(fileMap, path)
+	if err != nil {
+		log.Errorf("error downloading from NVD DB")
+	}
+
+	var allk8sCVEs, allistioCVEs []utils.NvdCVEEntry
+
+	for feedName, fileName := range feedReader {
+		dat, err := ioutil.ReadFile(fileName)
```

Comment:
Factor most of the loop body out into a function (the current function is already very long)